extensions/mongodb-client/deployment/src/main/java/io/quarkus/mongodb/deployment/MongoClientProcessor.java

```diff
@@ -147,10 +150,14 @@ CommandListenerBuildItem collectCommandListeners(CombinedIndexBuildItem indexBui
             MongoClientBuildTimeConfig buildTimeConfig, Capabilities capabilities) {
         Collection<ClassInfo> commandListenerClasses = indexBuildItem.getIndex()
                 .getAllKnownImplementors(DotName.createSimple(CommandListener.class.getName()));
-        List<String> names = commandListenerClasses.stream()
-                .map(ci -> ci.name().toString())
-                .collect(Collectors.toList());
-        return new CommandListenerBuildItem(names);
+        Stream<String> names = commandListenerClasses.stream()
+                .map(ci -> ci.name().toString());
+        Stream<String> tracing = Stream.empty();
+        if (buildTimeConfig.tracingEnabled && capabilities.isPresent(Capability.OPENTELEMETRY_TRACER)) {
+            tracing = Stream.of(MONGODB_TRACING_COMMANDLISTENER_CLASSNAME);
+        }
+        var items = Stream.concat(names, tracing).toList();
+        return new CommandListenerBuildItem(items);
```

Comment:
To avoid that you can index the `MongoTracingCommandListener`  from the deployment module of the extension, if tracing is enabled, so that is will be automatically discovered.

This is more "the quarkus way" ;)

Reply:
@loicmathieu  Thanks! By indexing you mean creating CommandListenerBuildItem in the opentelemetry deployment module? But I'd still have to check MongoClientBuildTimeConfig.tracingEnabled, right?

Reply:
`CommandListener` will be discoverable if in the Jandex index, by default extensions re not indexed so you need to emit an [AdditionalIndexedClassesBuildItem](https://github.com/quarkusio/quarkus/blob/main/core/deployment/src/main/java/io/quarkus/deployment/builditem/AdditionalIndexedClassesBuildItem.java) from the MongoDB extension if tracing is enabled and it will be automatically detected

Reply:
@loicmathieu I've pushed the change to to use [AdditionalIndexedClassesBuildItem](https://github.com/quarkusio/quarkus/blob/main/core/deployment/src/main/java/io/quarkus/deployment/builditem/AdditionalIndexedClassesBuildItem.java). I'm not sure about  the check on 
 `capabilities.isPresent(Capability.OPENTELEMETRY_TRACER)` Can it be left out and let the app fail at build time if the `quarkus.mongodb.tracing.enabled=true` but not opentelemetry tracing available?
Thanks!

Reply:
Let's keep it simple like that.